### PR TITLE
[Global] Better size validator messages + Refactor custom model error method

### LINKED ISSUE #82: Size validation message is not very good
Using:
```ruby
    validates :attachments, size: { less_than: 200.kilobytes }
```

The message I get is

> Attachments size 1.5 MB is not between required range

- This is broken English - numbers are "in" a range, not "between" a range
- This doesn't mention what the requirements are
- I didn't specify a range - just a maximum
- It should reference a singular "attachment"

Rails's numericality validator gives messages like this:

> Size must be less than 5

so something like

> Attachment size must be less than 200KB

 that would be good. (I know I can customize the message but I figure this would be a good improvement for anyone).

### PR DESCRIPTION
**Pull request by Mth0158**
Closes #82 <br>

This PR has 3 commits:

- [[SizeValidator] Better error messages depending on context, and make test coverage better](https://github.com/igorkasyanchuk/active_storage_validations/commit/142d3824d8c13b9bfd9650d188cebf9d5ac0f400) => fixes https://github.com/igorkasyanchuk/active_storage_validations/issues/82 + handles every test cases for size validator

I used Google translate for the translations, unsure it's the best translations though they seem pretty close to the previous one so if think it's okay?

- [[Validators] Make semantic changes for better understanding](https://github.com/igorkasyanchuk/active_storage_validations/commit/e9449b5880cb0a7af2adf29d6db53d4f22babf49)=> makes semantic changes for better understanding

- [[Global] Refactor custom model error method into a module thus making .with_message matcher available on all validators](https://github.com/igorkasyanchuk/active_storage_validations/commit/2490ce40e9280b49db737e744403485f2aa3f0ee) => shares the add_error method across validators, thus making the code easier to maintain + enable .with_message matcher on all validators

The second and third commit are optional, though I think it's a pretty nice addition that makes the code simple to understand and maintain + it makes the .with_message matcher available on all validators.

Open to discussion @gr8bit & @igorkasyanchuk

### CODE REVIEW

**Comment by Mth0158** [comment]
I rebased the PR onto master (plus added a missing part in the readme update for the 1st commit)

**Comment by Mth0158** [comment]
@gr8bit & @igorkasyanchuk let me know if I can do anything to help you on this review! :)

File: lib/active_storage_validations/aspect_ratio_validator.rb [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
 
     AVAILABLE_CHECKS = %i[with].freeze
     PRECISION = 3
 
     def check_validity!
-      return true if AVAILABLE_CHECKS.any? { |argument| options.key?(argument) }
-      raise ArgumentError, 'You must pass :with to the validator'
+      unless AVAILABLE_CHECKS.any? { |argument| options.key?(argument) }
+        raise ArgumentError, 'You must pass :with to the validator'
+      end
```
Comment (gr8bit) [inline lib/active_storage_validations/aspect_ratio_validator.rb:16]: Method does not return `true` on success anymore. Is that intended?

Comment (Mth0158) [reply lib/active_storage_validations/aspect_ratio_validator.rb:16]: Sorry for the late answer, I did not see your comment!
For this change, I mimicked Rails implementation of this method ([https://api.rubyonrails.org/classes/ActiveModel/EachValidator.html#method-i-check_validity-21](https://api.rubyonrails.org/classes/ActiveModel/EachValidator.html#method-i-check_validity-21))

It looks like that returning true is not a requirement, this method should only check if the arguments are valid raising an error if not valid. Therefore I took the liberty of removing it, let me know if I overstepped on that one!

Comment (Mth0158) [reply lib/active_storage_validations/aspect_ratio_validator.rb:16]: @gr8bit Should I change something on the PR? It's been running for quite some time already :'(

**Comment by igorkasyanchuk** [comment]
@gr8bit do you think we can release this PR and some others?

**Comment by Mth0158** [comment]
@igorkasyanchuk & @gr8bit still available to help you on this one, let me know!

**Comment by Mth0158** [comment]
@igorkasyanchuk & @gr8bit still available 😅

**Comment by Mth0158** [comment]
@igorkasyanchuk & @gr8bit still available to discuss on this PR, I would love to contribute more on this repo but this PR has been stuck since January :/

File: lib/active_storage_validations/aspect_ratio_validator.rb (hunk 1 above)
Comment (igorkasyanchuk) [reply lib/active_storage_validations/aspect_ratio_validator.rb:16]: probably this is all good, any result except ArgumentError is fine

**Comment by igorkasyanchuk** [comment]
@Mth0158 sorry for a really late response, I'm in general tend to merge your PR, don't see anything critical

can you please fix conflicts and let's give is a 2-3 days and I'll merge and release it

**Comment by Mth0158** [comment]
@igorkasyanchuk alright! I fixed the conflict in the README.md
I mistakenly rebased instead of creating a new commit (oups), I let you checked it all, let me know how I can help

**Comment by Mth0158** [comment]
@igorkasyanchuk @gr8bit

**Comment by igorkasyanchuk** [comment]
@Mth0158 Sorry for a long, very very long time, but now it's merged. Will release a new version today! Thank you for your work!

**Pull request merged**